### ISSUE TITLE
"authors" attribute not available in "this" or "resolve.my_package_name".

### ISSUE THREAD

**Issue by KelSolaar** [nerdvegas/rez#388]
Hi,

I was doing some tests with and saw that the `authors` attribute was not available from the `this` attribute, I would like to access from the `commands` one, for example:

```python
def commands():
    print(this.name)
    print(this.version)
    print(resolve.nuke.name)
    print(resolve.nuke.version)
    print(this.authors)  # Raising exception
    print(resolve.nuke.authors)  # Raising exception
```

Introspecting what was going on I noticed that `version` is in the `globals()` but not the others, is it expected?

```python
def commands():
    print(globals()['version'])
```

Cheers,

Thomas

**Comment by nerdvegas** [comment]
Hey Thomas,

The 'this' instance in commands section is currently not actually a Variant
instance, but a separate object that only provides a subset of what the
actual variant class does (see rex_bindings.py).

The thinking here was to avoid having package.py files that become
dependent on the rez version in use. However now I'm leaning more towards
just switching to 'this' as the actual Variant instance - it's public and
needs to keep API compatibility anyway, and maintaining these binding
classes is just extra work.

I'll do this as a dedicated release, keep your eye out for it. It
*shouldn't* cause any problems but it's a fairly sweeping change at the
same time. You'll then have access to 'authors' and whatever other
attributes exist on Variant (including arbitrary attributes).

Thx
A

**Comment by KelSolaar** [comment]
@nerdvegas: Great thank you! I'll definitely keep an eye.

**Comment by nerdvegas** [comment]
Hi,

This is available in latest release (2.10.0). I haven't removed the object that wraps the Variant instances, but I have added a getattr so that you have access to all the Variant attributes.

Closing this issue, let me know if you have issues.
